compiler/rustc_expand/src/proc_macro.rs

```diff
@@ -133,8 +133,11 @@ impl MultiItemModifier for ProcMacroDerive {
             }
         }
 
-        // fail if there have been errors emitted
-        if ecx.sess.parse_sess.span_diagnostic.err_count() > error_count_before {
+        // fail if there have been errors emitted,
+        // or if there's any unparsed tokens left.
+        if parser.token != token::Eof
```

Comment:
I think it would be useful to point at the token, similar to what we do when parsing a module:

https://github.com/rust-lang/rust/blob/5c0ca08c662399c1c864310d1a20867d3ab68027/compiler/rustc_parse/src/parser/item.rs#L64-L70

Reply:
Done!